### ISSUE TITLE
Unit test failures due to make-vector

### ISSUE THREAD

**Issue by rzubek** [rzubek/CSLisp#24]
The following unit tests are breaking due to pull request: https://github.com/rzubek/CSLisp/pull/14 

```csharp
CompileAndRun(ctx, "(make-vector 3)", "()");
CompileAndRun(ctx, "(make-vector '(1 2))", "(1 2)");
CompileAndRun(ctx, "(set-vector-element! (make-vector '(1 2) 0 0))", "(0 2)");
```

@furesoft Did you see these errors when you ran the unit tests, before submitting the pull request?

**Comment by furesoft** [comment]
no i didnt see it

**Comment by furesoft** [comment]
ahh its beacause the compileandrunmethod only accepts a string, but make-vector produces a list so wee need a method to check that

**Comment by rzubek** [comment]
Let's leave the second argument for CompileAndRun as it is. It's set up like that, so that it's also verifying that the printed representation doesn't change.

Instead let's add a new printed representation for vectors, and use that in the unit test. Just like closures have a string representation like `[Closure ...]`, for vectors, this could be something like `[Vector 1 2 3]`. This happens in `Val.Print()`. 

And then we could have `CompileAndRun(ctx, "(make-vector '(1 2))", "[Vector 1 2]");`

(But we don't want every List<Val> to be interpreted as a vector, right? Adding a new class Vector that subclasses List<Val> would help here.)

**Comment by furesoft** [comment]
thats right. i will do the class implementation

**Comment by rzubek** [comment]
Sounds good, please let me know about the pull request!

**Comment by rzubek** [comment]
Fixed in commit 3c19cbf9a7c9fdb310e634debef512c77e4511ab
